### PR TITLE
Register user

### LINKED ISSUE #9: Investigate user authentication options
## Introduction

While scaffolding is important, user authentication is going to be a vital and possibly blocking part as soon as the scaffolding is done. To be effective, we should be prepared by having the authentication system figured out beforehand.

A good first step is to decide what implementation we are going to use. Should we use something provided by Laravel, or perhaps use the already implemented system provided by Backpack? Maybe there are other, better options?

This issue purely addresses the investigation phase. The investigation may be a group effort, but it can also be carried out by an individual.

When the investigation is done, or when the investigating party decides that it is done, the results may be presented to the rest of the group. The goal is to make a proposal for a system/solution that the group is comfortable with.

Please think about the following during the investigation:
  - What options are there?
  - What are the main pros/cons of each option?
  - How well does each option meet our requirements?
  - How big an effort would implementing each option be?

## Requirements

To determine the requirements, please look at the [sitemap](https://lucid.app/lucidchart/a7dd714a-9779-4e92-98b8-90606e9d13e6/edit?s=772&name=docs&callback=close&v=1141&callback_type=back&page=0_0#?folder_id=home&browser=icon) as well as the [backlog](https://github.com/chas-academy/u05-imdb-clone-team4/projects/1).

Ideally, we want an implementation is able to perform the following at the very least:
 - Sign users in and out
 - Define and keep track of a scope for each user
 - Add user info (name, profile picture etc)
 - Integrate with Backpack

### PR DESCRIPTION
**Pull request by axelra82**
User registration is now in a separate form and more clear in the header. Newly registered users will be assigned a role of `user` by default.

On Heroku, `review apps` and `staging` _our admin registration is still open_, however, in production admin users can't register themselves, i.e. only an existing admin can promote a user to an admin role 👍

Closes #9 

To see some examples of using a data object on a specific page, checkout the correlation between:

`routes\web.php -> line 24`

`app\Http\Controllers\User\RegisteredUserController.php -> line 18-23`

`resources\views\user\index.blade.php`

I would also recommend taking a minut to inspect the files in `resources\views\layouts\` and `resources\views\navigation\` to get a look at a clean templating system 🙌 

**PS.** I took some time to go over the overall file structure in controllers, views etc and tried to clean it up a little. Just so we can have a clearer file structure (with folders for our sub-pages, controllers etc).

I've also done some re-structuring in the templating structure. This way we won't have to redeclare HTML on every page, but rather use a layout template (see `resources/views/layouts` and `resources/views/navigation` for current template)

99% of everything I've done today I learned from [the video I shared in slack this Friday (March 12th)](https://youtu.be/MFh0Fd7BsjE). I highly recommend it for anyone that wants a deeper understanding of the changes in this PR 🤓

### CODE REVIEW

File: resources/views/index.blade.php [hunk 1]
```diff
@@ -1,12 +1,6 @@
...
-    <meta http-equiv="X-UA-Compatible" content="IE=edge">
-    <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>Home</title>
-</head>
-<body>
-    <h1>Home</h1>
-</body>
-</html>
\ No newline at end of file
+@extends('layouts.app')
```
Comment (johannasjo) [inline resources/views/index.blade.php:1]: do we want to use this syntax or the x-component syntax (as in the footer branch)?

**Review by johannasjo — APPROVED** [review]
Looks good to me, we have double done some things (the getting rid of the HTML part on every page) but at this point we can just choose either and it will be good :)

File: resources/views/index.blade.php (hunk 1 above)
Comment (axelra82) [reply resources/views/index.blade.php:1]: From what I can tell the x-component still needs to have the html markup on every page? 🤔
While the include/yield templating only declares this once and then imports or extends the main layout 😃 

I might have missed something in the footer branch templating though... Let's take a closer look at this together during todays meeting so we can decide on which one we'll use moving forward 👍

**Review by Tuliany — APPROVED** [review]
LGTM 👍🏽

**Pull request merged**